packetbeat/beater/packetbeat.go

```diff
@@ -172,9 +170,9 @@ func (pb *packetbeat) Run(b *beat.Beat) error {
 		service.Stop()
 	}
 
-	waitShutdown := pb.cmdLineArgs.waitShutdown
-	if waitShutdown != nil && *waitShutdown > 0 {
-		time.Sleep(time.Duration(*waitShutdown) * time.Second)
+	timeout := b.Config.ShutdownTimeout
```

Comment:
You right, I move it back to `packetbeat`. This will make it more clearer as it's not handled in `libbeat` yet.